hr_employee_leave_quota/models/hr_employee_leave_quota.py

```diff
@@ -0,0 +1,172 @@
+
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError
+from datetime import timedelta
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    leave_type_ids = fields.One2many(
+        string="Holiday Quota",
+        comodel_name="hr.employee.leave.quota",
+        inverse_name="employee_id",
+    )
+    cancel_reason_id = fields.Many2one(
+        'leave.quota.cancel.reason',
+        string="Reason for cancellation",
+        readonly=True,
+        ondelete="restrict")
+
+
+class HrEmployeeLeaveQuota(models.Model):
+    _name = "hr.employee.leave.quota"
+
+    employee_id = fields.Many2one(
+        string="Employee",
+        comodel_name="hr.employee",
+        readonly=True,
+    )
+    leave_type = fields.Selection(
+        [('sick', 'Sick'), ('vacation', 'Vacation')],
+        string="Leave Type",
+        required=True,
+    )
+    quota = fields.Integer(
+        string="Quota",
+    )
+    used_days = fields.Integer(
+        string="Used (days)",
+    )
+    balance = fields.Integer(
+        string="Balance",
+        onchange="_onchange_balance",
+    )
+
+    @api.onchange('quota', 'used_days')
+    def _onchange_balance(self):
+        for rec in self:
+            rec.balance = rec.quota - rec.used_days
+
+
+class HrEmployeeLeaveRequest(models.Model):
+    _name = "hr.employee.leave.request"
+
+    employee_id = fields.Many2one(
+        string="Employee",
+        comodel_name="hr.employee",
+        default=lambda self: self._default_hr_employee(),
+        readonly=True,
+    )
+    approver_id = fields.Many2one(
+        'res.users',
+        string="Approve By",
+        readonly=True,
+    )
+    request_line_ids = fields.One2many(
+        'hr.employee.leave.request.line',
+        inverse_name="request_id",
+    )
+    state = fields.Selection(
+        [("draft", "Draft"), ("waiting approval", "Waiting Approval"),
+            ("approved", "Approved"), ("rejected", "Rejected")],
+        string="Status",
+        readonly=True,
+        index=True,
+        copy=False,
+        default="draft",
+        track_visibility="onchange",
+    )
+
+    def _default_hr_employee(self):
+        Employee = self.env["hr.employee"]
+        employee_id = Employee.search([("user_id", "=", self.env.user.id)])
```

Comment:
```suggestion
        employee = Employee.search([("user_id", "=", self.env.user.id)])
```
If the variable is object, try not to end with "_id".